src/hooks/useENS.ts

```diff
@@ -0,0 +1,35 @@
+import useSWR from 'swr'
+import { resolveAvatar, resolveName } from 'thirdweb/extensions/ens'
+
+import { useThirdwebAuthUser } from '@/hooks/useAuthUser'
+import { thirdwebClient } from '@/utils/shared/thirdwebClient'
+
+async function getData(address?: string) {
+  if (!address) {
+    return null
+  }
+
+  const ens = await resolveName({
+    client: thirdwebClient,
+    address,
+  })
+
+  const avatarUrl = ens
+    ? await resolveAvatar({
+        client: thirdwebClient,
+        name: ens,
+      })
+    : null
+
+  return {
+    ens,
+    avatarUrl,
+  }
+}
+
+export function useENS() {
+  const { user } = useThirdwebAuthUser()
+  const address = user?.address
+
+  return useSWR(address, async () => await getData(address))
```

Comment:
Not the optimal way to do this, we don't need to access `address` from the scope inside the fetcher function
```suggestion
  return useSWR(user?.address, getData)
```
What you pass to the first arg of `useSWR` is passed as arg to the fetcher fn, so just passing getData would access the value in the way `swr` intends

Reply:
Thanks for the suggestion! Updated!